client/src/utils.rs

```diff
@@ -150,13 +158,62 @@ pub fn compile_yul_contracts() {
 			continue;
 		}
 		let name = name_with_suffix.strip_suffix(".yul").unwrap();
-		// // compile it
-		let code = read_yul_data(&name);
+		// compile it
+		let code = read_yul_data(name);
 		let compiled_contract = compile_yul(&code);
-		write_bytes_data(compiled_contract, &name).unwrap();
+		write_bytes_data(compiled_contract, name).unwrap();
 	}
 }
 
+/// Construct the secret keys and public keys from the given raw data
+pub fn keyset_from_raw<const N: usize>(
+	sks_raw: [[&str; 2]; N],
+) -> (Vec<SecretKey>, Vec<PublicKey>) {
+	let mut sks = Vec::new();
+	let mut pks = Vec::new();
+
+	for sk_raw in sks_raw {
+		let sk0_raw = bs58::decode(sk_raw[0]).into_vec().unwrap();
+		let sk1_raw = bs58::decode(sk_raw[1]).into_vec().unwrap();
+
+		let mut sk0_bytes: [u8; 32] = [0; 32];
+		sk0_bytes.copy_from_slice(&sk0_raw);
+		let mut sk1_bytes: [u8; 32] = [0; 32];
+		sk1_bytes.copy_from_slice(&sk1_raw);
+
+		let sk = SecretKey::from_raw([sk0_bytes, sk1_bytes]);
+		let pk = sk.public();
+
+		sks.push(sk);
+		pks.push(pk);
+	}
+
+	(sks, pks)
+}
+
+/// Get the attestations from the contract
+pub async fn get_attestations(config: &ClientConfig) -> Result<Vec<Attestation>, EigenError> {
+	let client = setup_client(&config.mnemonic, &config.ethereum_node_url);
+	let contract = AttestationStation::new(config.as_address.parse::<Address>().unwrap(), client);
+	let events = contract.event::<AttestationCreatedFilter>().from_block(0);
+	let mut stream = events.stream().await.unwrap();
+	let mut attestations = Vec::new();
+
+	while let Ok(Some(Ok(att_created))) = timeout(Duration::from_secs(10), stream.next()).await {
```

Comment:
Timeouts are not ideal. Is there a way to fetch all historic events instead of streaming them?
I did some quick research, perhaps there is a solution here:
https://github.com/gakonst/ethers-rs/issues/988 or
https://github.com/gakonst/ethers-rs/blob/master/examples/events/examples/filtering.rs

Reply:
This has been very helpful! Thanks, just submitted a fix for it using `logs` instead of a `stream`